packages/@react-spectrum/s2/src/Calendar.tsx

```diff
@@ -10,38 +10,307 @@
+                display: 'flex',
+                flexDirection: 'row',
+                gap: 24,
+                width: 'full'
+              })}>
+              {Array.from({length: visibleMonths}).map((_, i) => (
+                <CalendarGrid offset={{months: i}} key={i}>
+                  <CalendarGridHeader>
+                    {(day) => (
+                      <CalendarHeaderCell className={headerCellStyles}>
+                        {day}
+                      </CalendarHeaderCell>
+                    )}
+                  </CalendarGridHeader>
+                  <CalendarGridBody>
+                    {(date) => (
+                      <CalendarCell date={date} />
+                    )}
+                  </CalendarGridBody>
+                </CalendarGrid>
+              ))}
+            </div>
+            {errorMessage && (
+              <Text slot="errorMessage" className={helpTextStyles({isInvalid, isDisabled})}>
+                {/* @ts-ignore */}
+                {errorMessage}
+              </Text>
+            )}
+          </>
+        );
+      }}
     </AriaCalendar>
   );
-}
+});
+
+// Ordinarily the heading is a formatted date range, ie January 2025 - February 2025.
+// However, we want to show each month individually.
+const CalendarHeading = () => {
+  let {visibleRange, timeZone} = useContext(CalendarStateContext) ?? {};
+  let era: any = getEraFormat(visibleRange?.start) || getEraFormat(visibleRange?.end);
+  let monthFormatter = useDateFormatter({
+    month: 'long',
+    year: 'numeric',
+    era,
+    calendar: visibleRange?.start.calendar.identifier,
+    timeZone
+  });
+  let months = useMemo(() => {
+    if (!visibleRange) {
+      return [];
+    }
+    let months: string[] = [];
+    for (let i = visibleRange.start; i.compare(visibleRange.end) <= 0; i = i.add({months: 1})) {
+      // TODO: account for the first week possibly overlapping, like with a custom 454 calendar.
+      // there has to be a better way to do this...
+      if (i.month === visibleRange.start.month) {
+        i = i.add({weeks: 1});
+      }
+      months.push(monthFormatter.format(i.toDate(timeZone!)));
+    }
+    return months;
+  }, [visibleRange, monthFormatter, timeZone]);
+
+  return (
+    <Heading styles={headingStyles}>
+      {months.map((month, i) => {
+        if (i === 0) {
+          return (
+            <Fragment key={month}>
+              <div className={titleStyles}>{month}</div>
+            </Fragment>
+          );
+        } else {
+          return (
+            <Fragment key={month}>
+              {/* Spacers to account for Next/Previous buttons and gap, spelled out to show the math */}
+              <div className={style({visibility: 'hidden', width: 32})} role="presentation" />
+              <div className={style({visibility: 'hidden', width: 24})} role="presentation" />
+              <div className={style({visibility: 'hidden', width: 32})} role="presentation" />
```

Comment:
you shouldn't need presentation I think, visibility: hidden already removes it from the a11y tree